### PR TITLE
- Create Add Transaction UI

### PR DESCRIPTION
**Pull request by danacealfonso**
# 📲 What #

* Transaction Add UI

# ✅ Acceptance criteria #

- [x ] Tested it locally?

# ⏰ TODO #

* Create transaction add UI

# 🔗 References #

https://user-images.githubusercontent.com/44434044/118463566-7ab71a00-b732-11eb-9e96-d3b38a8f8f27.mov

### CODE REVIEW

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class TransactionAddFragmentS1 : Fragment() {
+    lateinit var binding: FragmentTransactionAdd1Binding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = DataBindingUtil.inflate(inflater,
```
[the remark this answers is not in the document]
Comment (kalvinbdr) [reply app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt:21]: @danacealfonso You can use auto-format for this. For Android Studio in Mac, `Cmd`+`Option`+`L` is the shortcut.

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragment.kt [hunk 2]
```diff
@@ -6,24 +6,32 @@ import android.view.View
...
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        binding = DataBindingUtil.inflate(
-            inflater,
-            R.layout.fragment_transaction_details,
-            container,
-            false
-        )
+        binding = DataBindingUtil.inflate(inflater,
```
[the remark this answers is not in the document]
Comment (kalvinbdr) [reply app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragment.kt]: @danacealfonso You can use auto-format. For Android Studio in Mac, `Cmd`+`Option`+`L` is the hotkey for this.

[the remark this answers is not in the document]
Comment (danacealfonso) [reply app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragment.kt]: @kalvinbdr ok thanks

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -66,6 +65,11 @@
     <string name="lbl_transaction_date">Transaction Date</string>
     <string name="lbl_added_to_wallet">Added to Wallet</string>
     <string name="lbl_transaction_image">Transaction Image</string>
+    <string name="add_transaction">Add Transaction</string>
+    <string name="lbl_which_wallet_do_you_want_to_add">Which wallet do you want to add?</string>
+    <string name="title_add_transaction">Add Transaction</string>
+    <string name="next">Next</string>
+    <string name="add">Add</string>
```
Comment (MarvinBlanca) [inline app/src/main/res/values/strings.xml]: Add prefix to this string names ex. btn_next or lbl_next depends on what it's use is

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.kwartracker.android.R
+import com.kwartracker.android.databinding.FragmentTransactionAdd1Binding
+
+class TransactionAddFragmentS1 : Fragment() {
```
Comment (MarvinBlanca) [inline app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt]: Please give a clear and readable class name.

**Review by MarvinBlanca — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (danacealfonso) [reply app/src/main/res/values/strings.xml]: Revised in 1d68f15a12935763089100ada272a41f8f3c8fe9

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt (hunk 4 above)
Comment (danacealfonso) [reply app/src/main/java/com/kwartracker/android/transactions/ui/add/TransactionAddFragmentS1.kt]: Revised in 9dbd60c7fcc1434d18d03abebc2818a01d6f8cfc

**Review by MarvinBlanca — APPROVED** [review]

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        android:background="@color/content_main"
+        tools:context=".profile.ui.MyProfileFragment" >
+
+        <TextView
+            android:id="@+id/textView3"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls remove if not used else use snake case

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:text="Enter Amount"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls add text to strings.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+            android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: psl add color to colors.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+            android:text="₱"
+            android:textSize="50sp"
+            app:layout_constraintEnd_toStartOf="@+id/editText"
+            app:layout_constraintHorizontal_bias="1.0"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:ignore="ContentDescription|RelativeOverlap,HardcodedText" />
+
+        <EditText
+            android:id="@+id/editText"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls use snake case

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:text="00.00"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls add text to strings.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/editText">
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="20sp"
+                android:text="Transaction name"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls add text to strings.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+                android:id="@+id/et_trans_name"
+                android:layout_width="match_parent"
+                android:layout_height="55sp"
+                android:layout_marginLeft="20sp"
+                android:layout_marginTop="10sp"
+                android:layout_marginRight="20sp"
+                android:layout_marginBottom="20sp"
+                android:background="@drawable/edittext_bg_round_corner"
+                android:elevation="10sp"
+                android:hint="Select wallet to add"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls add text to strings.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+                android:inputType="text"
+                android:padding="14sp"
+                android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_2.xml]: pls add color to colors.xml

File: app/src/main/res/navigation/nav_graph.xml [hunk 13]
```diff
@@ -40,17 +40,47 @@
...
         tools:layout="@layout/fragment_transaction_add_1" />
+
+    <fragment
+        android:id="@+id/profile_fragment"
+        android:name="com.kwartracker.android.profile.ui.MyProfileFragment"
+        android:label="Transaction"
+        tools:layout="@layout/fragment_my_profile" />
+
+    <fragment
+        android:id="@+id/dashoardFragment"
```
Comment (rexendy) [inline app/src/main/res/navigation/nav_graph.xml]: pls use snake case

File: app/src/main/res/navigation/nav_graph.xml [hunk 14]
```diff
@@ -40,17 +40,47 @@
...
+        android:name="com.kwartracker.android.dashboard.ui.DashboardFragment"
+        android:label="Login"
```
Comment (rexendy) [inline app/src/main/res/navigation/nav_graph.xml]: pls add to strings.xml

File: app/src/main/res/navigation/nav_graph.xml [hunk 15]
```diff
@@ -40,17 +40,47 @@
...
+        tools:layout="@layout/fragment_dashboard" >
+        <action
+            android:id="@+id/action_dashoardFragment_to_walletsFragment"
+            app:destination="@id/walletsFragment" />
```
Comment (rexendy) [inline app/src/main/res/navigation/nav_graph.xml]: pls use snake case

**Review by rexendy — CHANGES_REQUESTED** [review]
- pls use snake case in id (xml layout) else if not used pls remove it
- pls add text and colors to strings .xml and colors.xml

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 5 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 6 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 8 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 9 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 11 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/navigation/nav_graph.xml (hunk 13 above)
Comment (danacealfonso) [reply app/src/main/res/navigation/nav_graph.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/navigation/nav_graph.xml (hunk 14 above)
Comment (danacealfonso) [reply app/src/main/res/navigation/nav_graph.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/navigation/nav_graph.xml (hunk 15 above)
Comment (danacealfonso) [reply app/src/main/res/navigation/nav_graph.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 7 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in d275fcc6e1f4b051f87db144c68e97987a9d6592

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 12 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in d275fcc6e1f4b051f87db144c68e97987a9d6592

File: app/src/main/res/layout/fragment_transaction_add_2.xml (hunk 10 above)
Comment (danacealfonso) [reply app/src/main/res/layout/fragment_transaction_add_2.xml]: fix in 49df7d5e62794cf0a6698424f963f76f5828af96

**Review by rexendy — APPROVED** [review]

**Review by jupilot11 — APPROVED** [review]

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/AddTransactionFragment.kt [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.kwartracker.android.R
+import com.kwartracker.android.databinding.FragmentTransactionAddBinding
+import com.kwartracker.android.transactions.ui.list.ListTransactionsFragment
+import com.kwartracker.android.transactions.ui.main.TransactionsFragment
+
+class AddTransactionFragment : Fragment() {
```
Comment (kalvinbdr) [inline app/src/main/java/com/kwartracker/android/transactions/ui/add/AddTransactionFragment.kt]: @danacealfonso 
Provide more comprehensible class name. What is the difference of this fragment to `AddNewTransactionFragment`?

File: app/src/main/res/navigation/nav_transactions.xml [hunk 17]
```diff
@@ -6,12 +6,12 @@
...
     <fragment
         android:id="@+id/transactions_details_fragment"
-        android:name="com.kwartracker.android.transactions.ui.details.TransactionDetailsFragment"
+        android:name="com.kwartracker.android.transactions.ui.details.DetailsTransactionFragment"
         android:label="Transaction Details"
         tools:layout="@layout/fragment_transaction_details" />
     <fragment
         android:id="@+id/transaction_lists_fragment"
-        android:name="com.kwartracker.android.transactions.ui.list.TransactionsListFragment"
+        android:name="com.kwartracker.android.transactions.ui.list.ListTransactionsFragment"
```
Comment (kalvinbdr) [inline app/src/main/res/navigation/nav_transactions.xml]: @danacealfonso 
```suggestion
        android:name="com.kwartracker.android.transactions.ui.list.TransactionListFragment"
```
Provide more comprehensible class name. It is best for class names in singular form. If it is a collection/list, name it to singular form + list.

File: app/src/main/res/values/colors.xml [hunk 18]
```diff
@@ -30,4 +30,8 @@
     <color name="light_green">#0EBCA7</color>
     <color name="light_red">#E26767</color>
 
+    <color name="transaction_field_name">#BBC3C9</color>
```
Comment (kalvinbdr) [inline app/src/main/res/values/colors.xml:33]: @danacealfonso 
Is this color used by all of the input fields?

File: app/src/main/res/navigation/nav_transactions.xml (hunk 17 above)
Comment (danacealfonso) [reply app/src/main/res/navigation/nav_transactions.xml]: fix in 8e6fa36a048fe726e08f61764cfae1f06a381b94

File: app/src/main/java/com/kwartracker/android/transactions/ui/add/AddTransactionFragment.kt (hunk 16 above)
Comment (danacealfonso) [reply app/src/main/java/com/kwartracker/android/transactions/ui/add/AddTransactionFragment.kt]: fix in 8e6fa36a048fe726e08f61764cfae1f06a381b94

File: app/src/main/res/values/colors.xml (hunk 18 above)
Comment (danacealfonso) [reply app/src/main/res/values/colors.xml:33]: it is used only in all transaction input field labels

**Review by kalvinbdr — APPROVED** [review]

**Pull request merged**